### DISCUSSION TITLE
Increase functionality of component ref

### DISCUSSION THREAD

**Discussion by KrzysztofMoch** [TheWidlarzGroup/react-native-video#4965] [Ideas]
### What problem are you trying to solve?

This will increase developer experience.

### Proposed solution

Add functions to the component ref to increase its functionality:

- A method to get the current playback position (time).
- A method to change the playback rate.
- A method to change the audio pan direction.
- A method to check if the video is a livestream or not.
- A method to control the volume/mute the video.
- A method to control/get the target latency.
- A method to get the live latency.

**Comment by alperpacin** [comment]
I second that. A method to control the volume/mute the video would be nice since setNativeProps is gone with @6.0.0-alpha.9.

```typescript
export interface VideoRef {
  seek: (time: number, tolerance?: number) => void;
  resume: () => void;
  pause: () => void;
  presentFullscreenPlayer: () => void;
  dismissFullscreenPlayer: () => void;
  restoreUserInterfaceForPictureInPictureStopCompleted: (
    restore: boolean,
  ) => void;
  save: (options: object) => Promise<VideoSaveData>;
}
```

**Comment by JustJoostNL** [comment]
@KrzysztofMoch 

Hey,
Will this be added soon? This is game changing for our use case.

We'd basically like all methods from [here](https://docs.expo.dev/versions/latest/sdk/av/#av), but if this feature gets prioritized, these should be done first:
- Play/pause
- get status
- set rate
- set volume/muted
- onPlaybackStatusUpdate

**Comment by alperpacin** [comment]
Any updates on this matter?

**Comment by KrzysztofMoch** [comment]
Hey, I'm on vacation until the end of the week, I'll see what I can do when I get back

**Comment by danielmark0116** [comment]
If there's a space, I could take this one and add these methods! 😎

**Comment by JustJoostNL** [comment]
@danielmark0116 that would be amazing!

**Comment by JustJoostNL** [comment]
@danielmark0116 any updates on this? 😄

**Comment by danielmark0116** [comment]
In the making! Aiming to finish it soon 🚀

**Comment by JustJoostNL** [comment]
Sick!

**Comment by nikonhub** [comment]
Wha'ts the reason the remove "setNativeProps" ? I used it to unset the src to limit the memory usage in an infinite list.

**Comment by KrzysztofMoch** [comment]
I have an issue with it to work in Interop Layer & it's deprecated so  I have decided to remove it - but if it will work for you, you are free to use it

**Comment by amitkumarcoding** [comment]
@KrzysztofMoch https://github.com/TheWidlarzGroup/react-native-video/issues/3843

**Comment by TWG-AI** [comment]
v7 exposes all of this through the player object returned by `useVideoPlayer`:

```ts
player.currentTime      // current position (seconds)
player.rate = 1.5       // playback speed
player.volume = 0.8     // volume 0–1
player.muted = true     // mute
player.play()
player.pause()
player.seekTo(30)       // seek to 30s
player.seekBy(-10)      // seek -10s from current
```

> I'm an AI — double-check the full API surface in the v7 docs.
